Hey @budhe888 
Thank you again for the commit, I tried applying it on top of the main d3c03448d1ff825b6204e34678f49938666b7aa0 
the PR applies fine. I usually check if the code works by running `make example`, and check the output of the resulting binary.

It seems one of the commits break the codebase
```
$ make example
gcc -Wall -Wextra -pedantic -std=c11 -Werror -Werror=strict-prototypes -Werror=format -Werror=old-style-definition -Werror=cast-align -I/home/alonso/projects/lpwan_examples/nanocbor/include -I/home/alonso/projects/lpwan_examples/zcbor/include/ -c src/coreconfManipulation.c -o obj/coreconfManipulation.o
gcc -Wall -Wextra -pedantic -std=c11 -Werror -Werror=strict-prototypes -Werror=format -Werror=old-style-definition -Werror=cast-align -I/home/alonso/projects/lpwan_examples/nanocbor/include -I/home/alonso/projects/lpwan_examples/zcbor/include/ -c src/coreconfTypes.c -o obj/coreconfTypes.o
gcc -Wall -Wextra -pedantic -std=c11 -Werror -Werror=strict-prototypes -Werror=format -Werror=old-style-definition -Werror=cast-align -I/home/alonso/projects/lpwan_examples/nanocbor/include -I/home/alonso/projects/lpwan_examples/zcbor/include/ -c src/hashmap.c -o obj/hashmap.o
gcc -Wall -Wextra -pedantic -std=c11 -Werror -Werror=strict-prototypes -Werror=format -Werror=old-style-definition -Werror=cast-align -I/home/alonso/projects/lpwan_examples/nanocbor/include -I/home/alonso/projects/lpwan_examples/zcbor/include/ -c src/serialization.c -o obj/serialization.o
src/serialization.c: In function ‘_parse_array’:
src/serialization.c:231:9: error: void value not ignored as it ought to be
  231 |     if (nanocbor_leave_container(value, &cborArrayValue) < 0) {
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/serialization.c: In function ‘_parse_map’:
src/serialization.c:265:9: error: void value not ignored as it ought to be
  265 |     if (nanocbor_leave_container(value, &map) < 0) {
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [Makefile:60: obj/serialization.o] Error 1
```

I think it comes from one of the PR which assumes nanocbor_leave_container returns an int, but instead it returns void.
```c                                                                                                                                                                                         
   void nanocbor_leave_container(nanocbor_value_t *it,                                                                                                                                        
                                 nanocbor_value_t *container);                                                                                                                                
 ```    

Which nanocbor version are you using?
Or if this was an untested line, we can simply remove the checks..

--- a/tools/generateCBORDumps.py
+++ b/tools/generateCBORDumps.py
@@ -1,3 +1,12 @@
+# /// script
+# requires-python = ">=3.13"
+# dependencies = [
+#     "cbor2",
+#     "jinja2",
+#     "pycoreconf @ git+ssh://git@github.com/budhe888/pycoreconf",
+# ]
+# ///
+
 # Example python3 generateCBORDumps.py ../samples/simple_yang/sensor_instance.json ../samples/simple_yang/sensor@unknown.sid cborDumpsHeader.h
 import cbor2
 import pycoreconf

--- a/tools/generateStubs.py
+++ b/tools/generateStubs.py
@@ -1,3 +1,12 @@
+# /// script
+# requires-python = ">=3.13"
+# dependencies = [
+#     "cbor2",
+#     "jinja2",
+#     "pycoreconf @ git+ssh://git@github.com/budhe888/pycoreconf",
+# ]
+# ///
+
 # Read a SID JSON file and Generate a C header file with stubs for all the SID functions
 
 # Import argparse and write a command line parser to accept .sid file as input

